analyzers/tests/SonarAnalyzer.UnitTest/MetadataReferences/NuGetMetadataFactory.cs

```diff
@@ -83,47 +83,26 @@ public static IEnumerable<MetadataReference> CreateNETStandard21()
                .ToImmutableArray();
         }
 
-        private static IEnumerable<MetadataReference> Create(Package package, string[] allowedTargetFrameworks)
+        private static IEnumerable<MetadataReference> Create(Package package, string[] allowedDirectories)
         {
+            Console.WriteLine();
+            Console.WriteLine($"Package: {package.Id}, {package.Version}");
             package.EnsurePackageIsInstalled();
 
-            var allowedNuGetLibDirectoriesByPreference = allowedTargetFrameworks.Select((folder, priority) => new { folder, priority });
             var packageDirectory = package.PackageDirectory();
-            LogMessage($"Download package directory: {packageDirectory}");
             if (!Directory.Exists(packageDirectory))
             {
-                throw new ApplicationException($"Test setup error: folder for downloaded package does not exist. Folder: {packageDirectory}");
+                throw new ApplicationException($"Test setup error: Package directory doesn't exist: {packageDirectory}");
             }
-
-            var matchingDllsGroups = Directory.GetFiles(packageDirectory, "*.dll", SearchOption.AllDirectories)
-                .Select(path => new FileInfo(path))
-                .GroupBy(file => file.Directory.Name).ToArray();
-            IGrouping<string, FileInfo> selectedGroup;
-
-            selectedGroup = matchingDllsGroups.Length == 1 && matchingDllsGroups[0].Key.EndsWith(".dll")
-                ? matchingDllsGroups[0]
-                : matchingDllsGroups.Join(allowedNuGetLibDirectoriesByPreference,
-                    group => group.Key.Split('+').First(),
-                    allowed => allowed.folder,
-                    (group, allowed) => new { group, allowed.priority })
-                .OrderBy(merged => merged.priority)
-                .First()
-                .group;
-
-            DumpSelectedGroup(package, selectedGroup);
-
-            return selectedGroup.Select(file => MetadataReference.CreateFromFile(file.FullName)).ToImmutableArray();
-        }
-
-        private static void DumpSelectedGroup(Package package, IGrouping<string, FileInfo> fileGroup)
-        {
-            Console.WriteLine();
-            Console.WriteLine($"Package: {package.Id}");
-            Console.WriteLine($"Version: {package.Version}, chosen targetFramework: {fileGroup.Key}");
-            foreach (var file in fileGroup)
+            var dllsPerDirectory = Directory.GetFiles(packageDirectory, "*.dll", SearchOption.AllDirectories)
+                .GroupBy(x => new FileInfo(x).Directory.Name)
+                .ToDictionary(x => x.Key.Split('+').First(), x => x.AsEnumerable());
+            var dlls = dllsPerDirectory[allowedDirectories.First(x => dllsPerDirectory.ContainsKey(x))];
```

Comment:
I suggest to use `dir` rather than `x` in the lambda because when reading `dllsPerDirectory.ContainsKey(x)` the brain can wonder if `x` is a `dll` or a `directory`

Reply:
I prefer `x` as a convention from mathematical lambda calculus. It's usually λx for single argument. Like this, you don't need to ask yourself if `dir` is a local variable, argument or field. It's whatever the lambda is run for.

(The `x` situation doesn't apply for nested lambda or with multiple arguments)

Reply:
maybe we should document this in our coding style